### PR TITLE
buildbot: 2.4.1 -> 2.5.0

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Updates buildbot to the latest version.

Buildbot has acquired some new test dependencies, including a circular dependency on the `buildbot_www` plugin, which requires using a wheel (as was done in the past).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @nand0p @ryansydnor

### CODE REVIEW

**Comment by lopsided98** [comment]
@GrahamcOfBorg test buildbot

**Comment by lopsided98** [comment]
`x86_64-linux` build always fails on ofborg.

**Comment by veprbl** [comment]
> `x86_64-linux` build always fails on ofborg.

Usually only the test was failing. This time it's the build fails on the tests that somehow are sensitive to a warning:
```
builtins.DeprecationWarning: The usage of `cmp` is deprecated and will be removed on or after 2021-06-01.  Please use `eq` and `order` instead.
```

This is not an issue for aarch64 because there we don't run tests:
https://github.com/NixOS/nixpkgs/blob/7edc64b63394d169ec9b9b2351c9b0f82f84e675/pkgs/development/python-modules/buildbot/default.nix#L81-L82

File: pkgs/development/python-modules/buildbot/plugins.nix [hunk 1]
```diff
@@ -6,11 +6,13 @@
...
     src = fetchPypi {
-      inherit pname version;
-      sha256 = "0l4kyxh62y86cw97101gjs42n1sdw1n18cgh6mm337gzjn42nv3x";
+      pname = "buildbot_www";
+      inherit version format;
+      python = "py3";
+      sha256 = "1wbkkv046h6yb0fnh39ypva4x809dphjwvzv43dpgpnshlya4aws";
     };
 
-    buildInputs = [ buildbot buildbot-pkg mock ];
```
Comment (veprbl) [inline pkgs/development/python-modules/buildbot/plugins.nix]: These should have been `checkInputs`? If yes, we could use `buildbot.overrideAttrs { doCheck = false; checkInputs = []; }`.

File: pkgs/development/python-modules/buildbot/plugins.nix [hunk 2]
```diff
@@ -6,11 +6,13 @@
...
+    format = "wheel";
```
Comment (FRidh) [inline pkgs/development/python-modules/buildbot/plugins.nix]: avoid wheels when possible

File: pkgs/development/python-modules/buildbot/plugins.nix (hunk 1 above)
Comment (lopsided98) [reply pkgs/development/python-modules/buildbot/plugins.nix]: No, if they are not available, the build fails with:
```Please install buildbot, buildbot_pkg, and mock modules in order to install that package, or use the pre-build .whl modules available on pypi```

See: https://github.com/buildbot/buildbot/blob/master/www/base/setup.py#L24

File: pkgs/development/python-modules/buildbot/plugins.nix (hunk 2 above)
Comment (lopsided98) [reply pkgs/development/python-modules/buildbot/plugins.nix]: The only way I can see to avoid a wheel is to patch the package's setuptools script.

Comment (FRidh) [reply pkgs/development/python-modules/buildbot/plugins.nix]: A comment explaining that should be added.
How about using `pipInstallFlags = [ "--ignore-dependencies" ];` instead?

**Comment by lopsided98** [comment]
The tests passed locally when I wrote this PR against unstable, but broke due to a dependency upgrade when rebased against master. The issue (https://github.com/glyph/automat/issues/117) has not been fixed upstream, so I applied a patch used by Arch Linux to ignore the warning for now.

I was not able to find a way to stop `buildbot-plugins.www` from depending on buildbot other than patching `setup.py` with `sed`, so that is what I ended up doing. Using `--ignore-dependencies` (or other flags) had no effect, even after removing the custom import check.

None of the plugins actually have any Python tests (`www` has Javascript tests that we don't run), so I simply set `doCheck = false` for all of them, removing the circular dependency. I was asked to do this in the past, but at the time thought it wasn't a good idea. The only reason `buildbot` was needed as a `checkInput` is that the `__init__.py` for each plugin imports it, and this is run even if there are no tests.

**Comment by lopsided98** [comment]
@GrahamcOfBorg test buildbot

**Comment by lopsided98** [comment]
@GrahamcOfBorg test buildbot

**Comment by FRidh** [comment]
Thanks for the explanation. Could you rebase your changes?

**Comment by jonringer** [comment]
@GrahamcOfBorg build buildbot-full

File: pkgs/development/python-modules/buildbot/pkg.nix [hunk 3]
```diff
@@ -6,7 +6,7 @@ buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/development/python-modules/buildbot/pkg.nix]:
```suggestion
  disabled = isPy27;

```
python2 no longer supported:

```
builder for '/nix/store/7x9nxc3f389kvymq39cxqsll1cydmm57-python2.7-buildbot-console-view-2.5.0.drv' failed with exit code 1; last 10 log lines:
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 24
      file=sys.stderr)
          ^
  SyntaxError: invalid syntax
builder for '/nix/store/ldpks4ascfznrrz1fb2qm77f1yriww9s-python2.7-buildbot-grid-view-2.5.0.drv' failed with exit code 1; last 10 log lines:
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 24
      file=sys.stderr)
          ^
  SyntaxError: invalid syntax
builder for '/nix/store/jakr8nz715y7ckqbjzzqpkdmgld7yy2m-python2.7-buildbot-wsgi-dashboards-2.5.0.drv' failed with exit code 1; last 10 log lines:
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 24
      file=sys.stderr)
          ^
  SyntaxError: invalid syntax
builder for '/nix/store/vyldijnxlp3r0x5lklsrvavygq814q6i-python2.7-buildbot-waterfall-view-2.5.0.drv' failed with exit code 1; last 10 log lines:
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 24
      file=sys.stderr)
          ^
  SyntaxError: invalid syntax
builder for '/nix/store/wsknkgp86ph08flxrc8sq3l5yv7iwjp8-python2.7-buildbot-www-2.5.0.drv' failed with exit code 1; last 10 log lines:
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 25
      'available on pypi', file=sys.stderr)
                               ^
  SyntaxError: invalid syntax
cannot build derivation '/nix/store/zd9qb53i2m1lwqynrisgbyds17a4lmcs-env.drv': 5 dependencies couldn't be built
[14 built (5 failed), 2 copied (0.0 MiB), 0.0 MiB DL]
error: build of '/nix/store/zd9qb53i2m1lwqynrisgbyds17a4lmcs-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/72059
5 package failed to build:
python27Packages.buildbot-plugins.console-view python27Packages.buildbot-plugins.grid-view python27Packages.buildbot-plugins.waterfall-view python27Packages.buildbot-plugins.wsgi-dashboards python27Packages.buildbot-plugins.www

13 package were build:
buildbot buildbot-full buildbot-ui python37Packages.buildbot-plugins.console-view python37Packages.buildbot-plugins.grid-view python37Packages.buildbot-plugins.waterfall-view python37Packages.buildbot-plugins.wsgi-dashboards python37Packages.buildbot-plugins.www python38Packages.buildbot-plugins.console-view python38Packages.buildbot-plugins.grid-view python38Packages.buildbot-plugins.waterfall-view python38Packages.buildbot-plugins.wsgi-dashboards python38Packages.buildbot-plugins.www
```

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executables seem to work
not sure how to test plugins

```
[4 built, 2 copied (0.7 MiB), 0.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72059
13 package were build:
buildbot buildbot-full buildbot-ui python37Packages.buildbot-plugins.console-view python37Packages.buildbot-plugins.grid-view python37Packages.buildbot-plugins.waterfall-view python37Packages.buildbot-plugins.wsgi-dashboards python37Packages.buildbot-plugins.www python38Packages.buildbot-plugins.console-view python38Packages.buildbot-plugins.grid-view python38Packages.buildbot-plugins.waterfall-view python38Packages.buildbot-plugins.wsgi-dashboards python38Packages.buildbot-plugins.www
```

**Comment by lopsided98** [comment]
Can this be merged?

**Comment by jonringer** [comment]
review still passes, merging :)

**Pull request merged**